components/CardContainer/CardContainer.js

```diff
@@ -84,7 +96,7 @@ export default function CardContainer({
       </StyledCardContainerQuestion>
 
       <StyledCardContainerAnswer onClick={flipCard} $level={level}>
-        {showCardMenu()}
+        {showCardMenu(answer)}
```

Comment:
@marielngr

While I understand the intention to avoid code duplication by using a function call, this approach introduces potential performance drawbacks that outweigh the benefits:

1. Function Redefinition: Each time the component renders, showCardMenu is redefined. This not only incurs unnecessary performance overhead but also prevents effective optimization of the JSX by React's reconciliation process.
2. JSX Optimization: Since showCardMenu is a function inside the component, React can't optimize the rendering of <CardMenu> effectively. This function creates a new instance of the JSX on every render, leading to potential inefficiencies.
3. Simplification: The code inside showCardMenu isn't complex enough to justify its separation into a function for the sake of avoiding duplication. Instead, directly embedding the <CardMenu> component in the JSX, as shown below, makes the code more straightforward and maintains optimal performance:


Please change it, as this is considered an anti-pattern.

Reply:
@doemser Wow, thanks for the good explanation, that makes a lot of sense. Good to know that common guidelines (do not repeat yourself) do not always hold for React.  😃
We will change that

Reply:
fixed.

Reply:
function has been deleted

Reply:
function has been deleted